File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/assignmentovertimeandremainingcosts.rb
module Asposetasksjava
  module AssignmentOvertimeAndRemainingCosts
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        resource_assignments = project.getResourceAssignments().toList()

        # Parse through all the collected resource assignments
        i = 0
        while i < resource_assignments.size()
          ra = resource_assignments.get(i)
          puts "Overtime Cost: " + ra.get(Rjb::import('com.aspose.tasks.Asn').OVERTIME_COST).toString()
          puts "Overtime Work: " + ra.get(Rjb::import('com.aspose.tasks.Asn').OVERTIME_WORK).toString()
          puts "Remaining Cost: " + ra.get(Rjb::import('com.aspose.tasks.Asn').REMAINING_COST).toString()
          puts "Remaining Overtime Cost: " + ra.get(Rjb::import('com.aspose.tasks.Asn').REMAINING_OVERTIME_COST).toString()
          puts "Remaining Overtime Work: " + ra.get(Rjb::import('com.aspose.tasks.Asn').REMAINING_OVERTIME_WORK).toString()
          puts "--------------------------------------------------------"
          i +=1
        end    
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/AssignmentOvertimeAndRemainingCosts.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Asn as Asn;
class AssignmentOvertimeAndRemainingCosts{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $resource_assignments = $project->getResourceAssignments()->toList();

        # Parse through all the collected resource assignments
        $asn=new Asn();
        $i = 0;
        while ($i < sizeof($resource_assignments)) {
          $ra = $resource_assignments -> get($i);
          print "Overtime Cost: " . $ra -> get($asn -> OVERTIME_COST).PHP_EOL;// . toString()
          print "Overtime Work: " . $ra -> get($asn -> OVERTIME_WORK).PHP_EOL;// . toString()
          print "Remaining Cost: " . $ra -> get($asn -> REMAINING_COST).PHP_EOL;// . toString()
          print "Remaining Overtime Cost: " . $ra -> get($asn -> REMAINING_OVERTIME_COST).PHP_EOL;// . toString()
          print "Remaining Overtime Work: " . $ra -> get($asn -> REMAINING_OVERTIME_WORK).PHP_EOL;// . toString()
          print "--------------------------------------------------------".PHP_EOL;
          $i += 1;
        }


    }

}
?>